src/test/java/org/jacp/competitionservice/CompetitionControllerIT.java

```diff
@@ -0,0 +1,146 @@
+        searchDto.setParticipantDto(List.of(participantDto));
+        searchDto.setDuration(10);
+
+        var questionDto = new QuestionDto();
+        questionDto.setId(1L);
+
+        String requestBody = objectMapper.writeValueAsString(searchDto);
+        when(questionService.getQuestionsByFilter(filterDto)).thenReturn(List.of(questionDto));
+
+        // when
+        var response = mockMvc.perform(MockMvcRequestBuilders.post("/api/v1/competition/create")
+                        .content(requestBody)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andReturn().getResponse().getContentAsString();
+        var competitionEntityResponse = objectMapper.readValue(response, CompetitionEntity.class);
+
+        // then
+        assertEquals(searchDto.getDuration(), competitionEntityResponse.getDuration());
+    }
+
+    @Test
+    public void startCompetition_ReturnsCompetitionDto() throws Exception {
+        // given
+        Long competitionId = 1L;
+
+        // when
+        var response = mockMvc.perform(MockMvcRequestBuilders.post("/api/v1/competition/{competitionId}/start", competitionId))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+
+        var competitionDtoResponse = objectMapper.readValue(response, CompetitionDto.class);
+
+        // then
+        assertEquals(Status.QUEUED.toString(), competitionDtoResponse.getStatus());
+    }
+
+    @Test
+    public void joinParticipant_Success() throws Exception {
+        // given
+        Long competitionId = 1L;
+        long participantId = 1L;
+
+        // when
+        var response = mockMvc.perform(MockMvcRequestBuilders.post("/api/v1/competition/{competitionId}/join",
+                                competitionId)
+                        .param("participantId", Long.toString(participantId)))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+
+        var competitionDtoResponse = objectMapper.readValue(response, CompetitionDto.class);
+
+        // then
+        assertEquals(participantId, competitionDtoResponse.getParticipants().get(0));
+    }
+
+    @Test
+    public void getAllByStatusCompetition_Success() throws Exception {
+        // given
+        var status = Status.QUEUED;
+
+        // when
+        var response = mockMvc.perform(MockMvcRequestBuilders.get("/api/v1/competition")
+                        .param("status", status.toString()))
+                .andExpect(status().isOk())
+                .andReturn().getResponse().getContentAsString();
+
+        var competitionDtoResponseList = objectMapper.readValue(response, new TypeReference<List<CompetitionDto>>() {
+        });
+        // then
+        assertEquals(3, competitionDtoResponseList.size());
```

Comment:
Does it make sense to check that the status of all competitions is equal to expected?

Reply:
Fixed